string object_name
bool data